### DISCUSSION TITLE
My telegram bot  is not working.

### DISCUSSION THREAD

**Discussion by 9jadev** [def-studio/telegraph#485] [Q&A]
My Telegram bot is not working. It's a Laravel 10 app. I set the webhook, but it still did not work.

My Laravel telegraph.php config
```

<?php

use DefStudio\Telegraph\Telegraph;

return [
    /*
     * Telegram api base url, it can be overridden
     * for self-hosted servers
     */
    'telegram_api_url' => 'https://api.telegram.org/',

    /*
     * Sets Telegraph messages default parse mode
     * allowed values: html|markdown|MarkdownV2
     */
    'default_parse_mode' => Telegraph::PARSE_HTML,

    /*
     * Sets the handler to be used when Telegraph
     * receives a new webhook call.
     *
     * For reference, see https://defstudio.github.io/telegraph/webhooks/overview
     */
    // 'webhook_handler' => DefStudio\Telegraph\Handlers\EmptyWebhookHandler::class,
    'webhook_handler' => App\Http\Controllers\CustomWebhookHandler::class,

    /*
     * Sets a custom domain when registering a webhook. This will allow a local telegram bot api server
     * to reach the webhook. Disabled by default
     *
     * For reference, see https://core.telegram.org/bots/api#using-a-local-bot-api-server
     */
    // 'custom_webhook_domain' => 'http://my.custom.domain',

    /*
     * If enabled, Telegraph dumps received
     * webhook messages to logs
     */
    'debug_mode' => true,

    /*
     * If enabled, unknown webhook commands are
     * reported as exception in application logs
     */
    'report_unknown_webhook_commands' => true,

    'security' => [
        /*
         * if enabled, allows callback queries from unregistered chats
         */
        'allow_callback_queries_from_unknown_chats' => false,

        /*
         * if enabled, allows messages and commands from unregistered chats
         */
        'allow_messages_from_unknown_chats' => false,

        /*
         * if enabled, store unknown chats as new TelegraphChat models
         */
        'store_unknown_chats_in_db' => false,
    ],

    /*
     * Set model class for both TelegraphBot and TelegraphChat,
     * to allow more customization.
     *
     * Bot model must be or extend `DefStudio\Telegraph\Models\TelegraphBot::class`
     * Chat model must be or extend `DefStudio\Telegraph\Models\TelegraphChat::class`
     */
    'models' => [
        'bot' => DefStudio\Telegraph\Models\TelegraphBot::class,
        'chat' => DefStudio\Telegraph\Models\TelegraphChat::class,
    ],

    'storage' => [
        /**
         * Default storage driver to be used for Telegraph data
         */
        'default' => 'file',

        'stores' => [
            'file' => [
                /**
                 * Telegraph cache driver to be used, must implement
                 * DefStudio\Telegraph\Contracts\StorageDriver contract
                 */
                'driver' => \DefStudio\Telegraph\Storage\FileStorageDriver::class,

                /*
                 * Laravel Storage disk to use. See /config/filesystems/disks for available disks
                 * If 'null', Laravel default store will be used,
                 */
                'disk' => 'local',

                /**
                 * Folder inside filesystem to be used as root for Telegraph storage
                 */
                'root' => 'telegraph',
            ],
            'cache' => [
                /**
                 * Telegraph cache driver to be used, must implement
                 * DefStudio\Telegraph\Contracts\StorageDriver contract
                 */
                'driver' => \DefStudio\Telegraph\Storage\CacheStorageDriver::class,

                /*
                 * Laravel Cache store to use. See /config/cache/stores for available stores
                 * If 'null', Laravel default store will be used,
                 */
                'store' => null,

                /*
                 * Prefix to be prepended to cache keys
                 */
                'key_prefix' => 'tgph',
            ],
        ],
    ],
];

```

CustomerWebhookHandler

`<?php

namespace App\Http\Controllers;
use DefStudio\Telegraph\Handlers\WebhookHandler;

use DefStudio\Telegraph\DTO\CallbackQuery;
use DefStudio\Telegraph\DTO\Chat;
use DefStudio\Telegraph\DTO\InlineQuery;
use DefStudio\Telegraph\DTO\Message;
use DefStudio\Telegraph\DTO\User;
use DefStudio\Telegraph\Exceptions\TelegramWebhookException;
use DefStudio\Telegraph\Keyboard\Keyboard;
use DefStudio\Telegraph\Models\TelegraphBot;
use DefStudio\Telegraph\Models\TelegraphChat;
use Illuminate\Http\Request;
use Illuminate\Support\Collection;
use Illuminate\Support\Facades\Log;
use Illuminate\Support\Str;
use Illuminate\Support\Stringable;
use ReflectionMethod;
use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;

class CustomWebhookHandler extends WebhookHandler
{
    protected function handleChatMessage(Stringable $text): void
    {
        // checks that the contact matches the sender ID
        logs()->info($this->chat);

        $contactUserId = $this->message->contact()->userId();
        $senderId = $this->message->from()->id();

        if($contactUserId == $senderId){
            $this->reply('this is your contact');
        }else{
            $this->reply('this is NOT your contact');
        }
    }

    // public function chatid(): void
    // {
    //     logs()->info("bihk bmbmb");
    //     logs()->info($this->chat);
    //     $this->chat->html("Chat ID: {$this->chat->chat_id}")->send();
    // }

    public function handle(Request $request, TelegraphBot $bot): void
    {
        $this->bot = $bot;

        $this->request = $request;

        logs()->info($request);

        if ($this->request->has('message')) {
            /* @phpstan-ignore-next-line */
            logs()->info(22345);
            $this->message = Message::fromArray($this->request->input('message'));
            $this->handleMessage();

            return;
        }

        if ($this->request->has('edited_message')) {
            /* @phpstan-ignore-next-line */
            $this->message = Message::fromArray($this->request->input('edited_message'));
            $this->handleMessage();

            return;
        }

        // if ($this->request->has('channel_post')) {
        //     /* @phpstan-ignore-next-line */
        //     $this->message = Message::fromArray($this->request->input('channel_post'));
        //     $this->handleMessage();

        //     return;
        // }

        // if ($this->request->has('callback_query')) {
        //     /* @phpstan-ignore-next-line */
        //     $this->callbackQuery = CallbackQuery::fromArray($this->request->input('callback_query'));
        //     $this->handleCallbackQuery();
        // }

        if ($this->request->has('inline_query')) {
            /* @phpstan-ignore-next-line */
            $this->handleInlineQuery(InlineQuery::fromArray($this->request->input('inline_query')));
        }
    }

    private function handleMessage(): void
    {
        $this->extractMessageData();

        if (config('telegraph.debug_mode')) {
            Log::debug('Telegraph webhook message', $this->data->toArray());
        }

        $text = Str::of($this->message?->text() ?? '');

        // if ($text->startsWith('/')) {
        //     $this->handleCommand($text);

        //     return;
        // }

        if ($this->message?->newChatMembers()->isNotEmpty()) {
            foreach ($this->message->newChatMembers() as $member) {
                $this->handleChatMemberJoined($member);
            }

            return;
        }

        if ($this->message?->leftChatMember() !== null) {
            $this->handleChatMemberLeft($this->message->leftChatMember());

            return;
        }

        $this->handleChatMessage($text);
    }
}
`

The logs were not working

**Comment by 9jadev** [comment]
the  error i get here is https://api.telegram.org/{bot}/getWebhookInfo

{
  "ok": true,
  "result": {
    "url": "https://backendapi.wakacommunity.com/public/telegraph/{bot}/webhook",
    "has_custom_certificate": false,
    "pending_update_count": 18,
    "last_error_date": 1705125268,
    "last_error_message": "Wrong response from the webhook: 404 Not Found",
    "max_connections": 40,
    "ip_address": "15.204.3.146"
  }
}

**Comment by fabio-ivona** [comment]
Have you enabled webooks for your bot?

**Comment by 9jadev** [comment]
Yes I have 
I also checked to confirm it from telegram bot docs

**Comment by fabio-ivona** [comment]
Can you please try to put some logs inside the base handler to check the point where it try to retrieve the bot from its ID?

**Comment by 9jadev** [comment]
Same thing

**Comment by fabio-ivona** [comment]
@9jadev I can't help you without more info, please set up a sample repository or more log info, such as `Telegraph::getEbhookDebugInfo()->send()->dump()` result
